src/Bicep.Core.IntegrationTests/Emit/TemplateEmitterTests.cs

```diff
@@ -104,7 +104,8 @@ public void ValidBicepTextWriter_TemplateEmiterTemplateHashCheck(DataSet dataSet
 
             // emitting the template should be successful
             var result = this.EmitTemplate(SyntaxTreeGroupingBuilder.Build(new FileResolver(), new Workspace(), PathHelper.FilePathToFileUrl(bicepFilePath)), memoryStream, ThisAssembly.AssemblyFileVersion);
-            result.Diagnostics.Should().BeEmpty();
+            // TODO: remove Where when the the support of parameter modifiers is dropped.
+            result.Diagnostics.Where(d => d.Code != "BCP153").Should().BeEmpty();
```

Comment:
>Where [](start = 31, length = 5)

I wonder if there's any value in creating a helper class (`ParameterModifierDeprecationHelper`?) that contains all the special test cases for this migration. Later on, we can just delete the class and follow the compiler errors to clean up.

Reply:
Yeah makes. I found that there's already the `DiagnosticCollectionExtensions` class so I added an extension method to it, which might be simpler than adding another helper class.